### PR TITLE
Add mock helpers to make working checking mock calls easier

### PR DESCRIPTION
**Pull request by mouadino**
First step toward better testing suite.

### CODE REVIEW

**Comment by bilunyk** [comment]
:+1: hamcrest is pretty sweet

**Comment by mouadino** [comment]
@bilunyk It's quite good, but writting custom matchers is hard to do specially getting error description right http://pyhamcrest.readthedocs.org/en/release-1.8/custom_matchers/ or maybe I am just too stupid to figure out how to do it properly :)

File: lymph/tests/test_mock_helpers.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        self.dummy_case = DummyTestCase()
+
+    def test_single_call_match(self):
+        self.dummy_case._assert_equal_calls(
+            [mock.call('func', 1, foo='bar')],
+            [mock.call('func', 1, foo='bar')]
+        )
+
+    def test_single_call_args_mismatch(self):
+        with self.assertRaises(AssertionError):
```
Comment (mamachanko) [inline lymph/tests/test_mock_helpers.py]: Did you consider asserting with `self.assertRaisesRegexp` ot see whether the correct message is included?

Comment (mouadino) [inline lymph/tests/test_mock_helpers.py]: Yes, I did but the error message are a work in progress, for keyword name you get a better error message than before, but not good enough, this is where I was thinking hamcrest matchers may be useful but then I gave up on them for now, but definitely when we have good enough message we can check them in our tests.

Comment (mamachanko) [inline lymph/tests/test_mock_helpers.py]: understood

Comment (mouadino) [inline lymph/tests/test_mock_helpers.py]: Correction: I mean hamcrest **custom** matchers, the idea is still to use the given matchers ofc :)

**Pull request merged**